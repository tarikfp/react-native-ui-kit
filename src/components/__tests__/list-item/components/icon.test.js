import * as React from "react";
import renderer from "react-test-renderer";
import ListItemIcon from "../../../list-item/components/icon";

describe("tests list item with icon component", () => {
  it("renders list item icon with name and type", () => {
    const tree = renderer
      .create(<ListItemIcon name="arrow-left" type="MaterialCommunityIcons" />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("renders list item icon with size", () => {
    const tree = renderer
      .create(
        <ListItemIcon
          name="arrow-left"
          type="MaterialCommunityIcons"
          size={45}
        />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("renders list item icon with color", () => {
    const tree = renderer
      .create(
        <ListItemIcon
          name="arrow-left"
          type="MaterialCommunityIcons"
          color="orange"
        />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("renders list item icon with backgroundColor", () => {
    const tree = renderer
      .create(
        <ListItemIcon
          name="arrow-left"
          type="MaterialCommunityIcons"
          backgroundColor="royalblue"
        />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("renders list item icon custom style", () => {
    const tree = renderer
      .create(
        <ListItemIcon
          name="arrow-left"
          type="MaterialCommunityIcons"
          style={{ marginLeft: 8 }}
        />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("renders list item icon custom wrapperStyle", () => {
    const tree = renderer
      .create(
        <ListItemIcon
          name="arrow-left"
          type="MaterialCommunityIcons"
          wrapperStyle={{ marginLeft: 8 }}
        />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("renders list item icon custom alignment", () => {
    const tree = renderer
      .create(
        <ListItemIcon
          name="arrow-left"
          type="MaterialCommunityIcons"
          alignment="start"
        />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("renders icon with custom icon props", () => {
    const tree = renderer
      .create(
        <ListItemIcon
          iconProps={{ name: "arrow-left", type: "MaterialCommunityIcons" }}
        />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
