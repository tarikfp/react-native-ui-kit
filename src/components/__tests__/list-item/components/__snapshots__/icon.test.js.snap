// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests list item with icon component renders icon with custom icon props 1`] = `
<View
  style={
    Array [
      Object {
        "flex": 0,
        "flexDirection": "row",
        "justifyContent": "center",
      },
      undefined,
    ]
  }
>
  <Text
    allowFontScaling={false}
    iconProps={
      Object {
        "name": "arrow-left",
        "type": "MaterialCommunityIcons",
      }
    }
    selectable={false}
    style={
      Array [
        Object {
          "color": "#757575",
          "fontSize": 20,
        },
        undefined,
        Object {
          "fontFamily": "FontAwesome",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        Object {},
      ]
    }
  >
    
  </Text>
</View>
`;

exports[`tests list item with icon component renders list item icon custom alignment 1`] = `
<View
  style={
    Array [
      Object {
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "flex-start",
      },
      undefined,
    ]
  }
>
  <Text
    allowFontScaling={false}
    selectable={false}
    style={
      Array [
        Object {
          "color": "#757575",
          "fontSize": 20,
        },
        undefined,
        Object {
          "fontFamily": "Material Design Icons",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        Object {},
      ]
    }
  >
    󰁍
  </Text>
</View>
`;

exports[`tests list item with icon component renders list item icon custom style 1`] = `
<View
  style={
    Array [
      Object {
        "flex": 0,
        "flexDirection": "row",
        "justifyContent": "center",
      },
      undefined,
    ]
  }
>
  <Text
    allowFontScaling={false}
    selectable={false}
    style={
      Array [
        Object {
          "color": "#757575",
          "fontSize": 20,
        },
        Object {
          "marginLeft": 8,
        },
        Object {
          "fontFamily": "Material Design Icons",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        Object {},
      ]
    }
  >
    󰁍
  </Text>
</View>
`;

exports[`tests list item with icon component renders list item icon custom wrapperStyle 1`] = `
<View
  style={
    Array [
      Object {
        "flex": 0,
        "flexDirection": "row",
        "justifyContent": "center",
      },
      Object {
        "marginLeft": 8,
      },
    ]
  }
>
  <Text
    allowFontScaling={false}
    selectable={false}
    style={
      Array [
        Object {
          "color": "#757575",
          "fontSize": 20,
        },
        undefined,
        Object {
          "fontFamily": "Material Design Icons",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        Object {},
      ]
    }
  >
    󰁍
  </Text>
</View>
`;

exports[`tests list item with icon component renders list item icon with backgroundColor 1`] = `
<View
  style={
    Array [
      Object {
        "flex": 0,
        "flexDirection": "row",
        "justifyContent": "center",
      },
      undefined,
    ]
  }
>
  <Text
    allowFontScaling={false}
    backgroundColor="royalblue"
    selectable={false}
    style={
      Array [
        Object {
          "color": "#757575",
          "fontSize": 20,
        },
        undefined,
        Object {
          "fontFamily": "Material Design Icons",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        Object {},
      ]
    }
  >
    󰁍
  </Text>
</View>
`;

exports[`tests list item with icon component renders list item icon with color 1`] = `
<View
  style={
    Array [
      Object {
        "flex": 0,
        "flexDirection": "row",
        "justifyContent": "center",
      },
      undefined,
    ]
  }
>
  <Text
    allowFontScaling={false}
    selectable={false}
    style={
      Array [
        Object {
          "color": "orange",
          "fontSize": 20,
        },
        undefined,
        Object {
          "fontFamily": "Material Design Icons",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        Object {},
      ]
    }
  >
    󰁍
  </Text>
</View>
`;

exports[`tests list item with icon component renders list item icon with name and type 1`] = `
<View
  style={
    Array [
      Object {
        "flex": 0,
        "flexDirection": "row",
        "justifyContent": "center",
      },
      undefined,
    ]
  }
>
  <Text
    allowFontScaling={false}
    selectable={false}
    style={
      Array [
        Object {
          "color": "#757575",
          "fontSize": 20,
        },
        undefined,
        Object {
          "fontFamily": "Material Design Icons",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        Object {},
      ]
    }
  >
    󰁍
  </Text>
</View>
`;

exports[`tests list item with icon component renders list item icon with size 1`] = `
<View
  style={
    Array [
      Object {
        "flex": 0,
        "flexDirection": "row",
        "justifyContent": "center",
      },
      undefined,
    ]
  }
>
  <Text
    allowFontScaling={false}
    selectable={false}
    style={
      Array [
        Object {
          "color": "#757575",
          "fontSize": 45,
        },
        undefined,
        Object {
          "fontFamily": "Material Design Icons",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        Object {},
      ]
    }
  >
    󰁍
  </Text>
</View>
`;
